pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -45,9 +48,32 @@ type Scheduler struct {
 	daemonOverhead map[*v1alpha5.Provisioner]v1.ResourceList
 	preferences    *Preferences
 	topology       *Topology
+	cluster        *state.Cluster
+	inflight       []*InFlightNode
 }
 
-func (s *Scheduler) Solve(ctx context.Context, pods []*v1.Pod) ([]*Node, error) {
+func (s *Scheduler) Solve(ctx context.Context, pods []*v1.Pod) ([]*Node, []*InFlightNode, error) {
+	provisionerMap := map[string]*v1alpha5.Provisioner{}
+	for _, provisioner := range s.provisioners {
+		provisionerMap[provisioner.Name] = provisioner
+	}
+
+	// create our in-flight nodes
+	s.cluster.ForEachNode(func(node *state.Node) bool {
+		provisionerName, ok := node.Node.Labels[v1alpha5.ProvisionerNameLabelKey]
+		if !ok {
+			// ignoring this node as it wasn't launched by us
+			return true
+		}
+		provisioner, ok := provisionerMap[provisionerName]
+		if !ok {
+			// ignoring this node as it wasn't launched by a provisioner that we recognize
+			return true
+		}
+		s.inflight = append(s.inflight, NewInFlightNode(node, s.topology, provisioner.Spec.StartupTaints, s.daemonOverhead[provisioner]))
+		return true
```

Comment:
If an instance doesn't launch because of an ICE or because the EC2 service is unschedulable, wouldn't we end up considering this node object as `inflight` ? Should we only be doing this once we've kicked off the node, rather than after we've completed successful scheduling?

Reply:
This shouldn't occur, we get a success from the instance launch before we create the node object.  The inflight nodes here are ones that were created in previous rounds of scheduling.

There is a scenario where a node fails to become ready and we think it will support pods.  It may prevent provisioning whatever excess capacity it has for 15 minutes, but it will resolve.  The same issue occurs currently where we bind the pods to the node that never comes up and they don't retry for 15 minutes.